src/ui.cpp

```diff
@@ -746,9 +746,10 @@ void uilist::calc_data()
     calculated_menu_size.x += calculated_hotkey_width + padding;
     calculated_menu_size.x += calculated_label_width + padding;
     calculated_menu_size.x += calculated_secondary_width + padding;
-    float max_avail_height = ImGui::GetMainViewport()->Size.y;
-    if( desired_bounds.has_value() ) {
-        max_avail_height = desired_bounds.value().h;
+    float max_avail_height = 0.9 * ImGui::GetMainViewport()->Size.y;
+    float desired_height;
+    if( desired_bounds.has_value() && ( desired_height = desired_bounds.value().h != -1 ) ) {
```

Comment:
Could you change `desired_height = desired_bounds.value().h != -1` please? It is confusing to me.
I assume it does:
1. eval `!=`
2. assign the resulting boolean value to the float desired_height

But I am not sure and I don't understand what it is for if it does that.

No, I guess it does:
1. assign `=`
2. compare `!=`
3. return compared value to the rest of the and statement.

No, the precedence of != is 10 and precedence of = is 16, so it has to first do !=, right?

Anyway, change it for us, lesser programmers, please :)

Reply:
Sure, I can do that.